lib/api/BackbeatAPI.js

```diff
@@ -339,6 +347,144 @@ class BackbeatAPI {
+                response.push({ bucket, key, versionId, site });
+            }
+            return cb(null, response);
+        });
+    }
+
+    /**
+     * Get all CRR operations that have failed.
+     * @param {Object} details - The route details
+     * @param {Function} cb - The callback to call
+     * @return {undefined}
+     */
+    getAllFailedCRR(details, cb) {
+        const cmds = ['hgetall', redisKeys.failedCRR];
+        this._redisClient.batch([cmds], (err, res) => {
+            if (err) {
+                return cb(err);
+            }
+            const [cmdErr, hashes] = res[0];
+            if (cmdErr) {
+                return cb(cmdErr);
+            }
+            const hashKeys = hashes ? Object.keys(hashes) : [];
+            const response = hashKeys.map(hashKey => {
+                const [bucket, key, versionId, site] = hashKey.split(':');
+                return { bucket, key, versionId, site };
+            });
+            return cb(null, response);
+        });
+    }
+
+    /**
+     * Retry all CRR operations that have failed.
+     * @param {Object} details - The route details
+     * @param {String} body - The POST request body string
+     * @param {Function} cb - The callback to call
+     * @return {undefined}
+     */
+    retryFailedCRR(details, body, cb) {
+        const { err, reqBody } = this._parseRetryFailedCRR(body);
+        if (err) {
+            return cb(err);
+        }
+        const fields = reqBody.map(o => {
+            const { bucket, key, versionId, site } = o;
+            return `${bucket}:${key}:${versionId}:${site}`;
+        });
+        const cmds = ['hmget', redisKeys.failedCRR, ...fields];
+        return this._redisClient.batch([cmds], (err, res) => {
+            if (err) {
+                return cb(err);
+            }
+            const [cmdErr, results] = res[0];
+            if (cmdErr) {
+                return cb(cmdErr);
+            }
+            // TODO: For all object metadata in results, push to kafka topics.
+            const response = [];
+            for (let i = 0; i < results.length; i++) {
+                if (results[i]) {
+                    const { bucket, key, versionId, site } = reqBody[i];
+                    const status = 'PENDING';
+                    response.push({ bucket, key, versionId, site, status });
+                }
+            }
+            return cb(null, response);
+        });
+    }
+
+    /**
+     * Validate that the POST request body has the necessary content.
+     * @param {String} body - The POST request body string
+     * @param {Function} cb - The callback to call
+     * @return {undefined}
+     */
+    _parseRetryFailedCRR(body) {
+        const msg = 'The body of your POST request is not well-formed';
+        let reqBody;
+        try {
+            reqBody = JSON.parse(body);
+        } catch (e) {
+            return {
+                err: errors.MalformedPOSTRequest.customizeDescription(msg),
```

Comment:
minor: it seems more common to return an `error` field in the response object than `err`.

Reply:
Updated.